apollo-router/src/axum_factory/compression/mod.rs

```diff
@@ -0,0 +1,186 @@
+                    Err(e) => {
+                        if let Err(_) = tx.send(Err(e.into())).await {
+                            return;
+                        }
+                    }
+                    Ok(data) => {
+                        let mut buf = BytesMut::zeroed(1024);
+                        let mut written = 0usize;
+
+                        let mut partial_input = PartialBuffer::new(&*data);
+                        loop {
+                            let mut partial_output = PartialBuffer::new(&mut buf);
+                            partial_output.advance(written);
+
+                            match self.encode(&mut partial_input, &mut partial_output) {
+                                Err(e) => panic!("{e:?}"),
+                                Ok(()) => {}
+                            }
+
+                            //let read = partial_input.written().len();
+                            written += partial_output.written().len();
+                            //println!("encode: read from input: {read}, written = {written}");
+
+                            if !partial_input.unwritten().is_empty() {
+                                // there was not enough space in the output buffer to compress everything,
+                                // so we resize and add more data
+                                if partial_output.unwritten().is_empty() {
+                                    let _ = partial_output.into_inner();
+                                    buf.reserve(written);
+                                }
+                            } else {
+                                // FIXME: what happens if we try to flush in a full buffer
+                                match self.flush(&mut partial_output) {
+                                    Err(e) => panic!("{e:?}"),
+                                    Ok(_) => {
+                                        let len = partial_output.written().len();
+                                        //println!("flush(b={b}) with buffer of size {len}");
+                                        let _ = partial_output.into_inner();
+                                        buf.resize(len, 0);
+                                        if let Err(_) = tx.send(Ok(buf.freeze())).await {
+                                            return;
+                                        }
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            let buf = BytesMut::zeroed(64);
+            let mut partial_output = PartialBuffer::new(buf);
+
+            match self.finish(&mut partial_output) {
+                Err(e) => panic!("{e:?}"),
+                Ok(_) => {
+                    let len = partial_output.written().len();
+                    //println!("finish with buffer of size {}", len);
```

Comment:
```suggestion
```

Reply:
fixed in [0fb0ba9](https://github.com/apollographql/router/pull/2986/commits/0fb0ba985404c3652c257bed12044f955b154d53)